api/scpca_portal/utils.py

```diff
@@ -50,3 +59,52 @@ def filter_dict_list_by_keys(list_of_dicts, included_keys, *, ignore_value_error
         new_list_of_dicts.append(dictionary_copy)
 
     return new_list_of_dicts
+
+
+def get_key_superset_from_dicts(list_of_dicts: List[Dict]) -> Set:
+    """Extracts each provided dictionary's key set and returns the superset of all key sets."""
+    key_superset = set()
+    for dictionary in list_of_dicts:
+        key_superset.update(set(dictionary.keys()))
+
+    return key_superset
+
+
+def write_dicts_to_file(list_of_dicts: List[Dict], output_file_path: str, **kwargs) -> None:
+    """
+    Writes a list of dictionaries to a csv-like file.
+    Optional modifiers to the csv.DictWriter can be passed to function as kwargs.
+    """
+    kwargs["fieldnames"] = kwargs.get("fieldnames", get_key_superset_from_dicts(list_of_dicts))
+    kwargs["delimiter"] = kwargs.get("delimiter", common.TAB)
+
+    with open(output_file_path, "w", newline="") as raw_file:
+        csv_writer = csv.DictWriter(raw_file, **kwargs)
+        csv_writer.writeheader()
+        csv_writer.writerows(list_of_dicts)
+
+
+def get_csv_zipped_values(
```

Comment:
1. We should remove the `model_name` argument as it is only used for error messaging.
2. Don't catch the exception just to log it, we want to fail here as it is indicative of malformed input data which means we probably don't want to process this csv file.
3. zipped values is an empty list but returns a zip object when successful this should just be either a list returned or a zip object.

Going to respond to an anticipated response being, we uniformly handle the errors so this would be less repetitive. I agree with that but it still think the caller should handle the error. Util functions shouldn't be logging, just throwing an expressive error so throwing a ValueError, *args are of non-equal lengths would be sufficient. The caller can decide to log, catch the exception, neither or both.